pkg/registry/thirdpartyresource/etcd/etcd_test.go

```diff
@@ -80,26 +80,24 @@ func TestCreate(t *testing.T) {
 func TestUpdate(t *testing.T) {
 	storage, fakeClient := newStorage(t)
 	test := resttest.New(t, storage, fakeClient.SetError)
-	key, err := storage.KeyFunc(test.TestContext(), "foo")
-	if err != nil {
-		t.Fatal(err)
-	}
-	key = etcdtest.AddPrefix(key)
-	fakeClient.ExpectNotFoundGet(key)
-	fakeClient.ChangeIndex = 2
-	rsrc := validNewThirdPartyResource("foo")
-	existing := validNewThirdPartyResource("exists")
-	existing.Namespace = test.TestNamespace()
-	obj, err := storage.Create(test.TestContext(), existing)
-	if err != nil {
-		t.Fatalf("unable to create object: %v", err)
-	}
-	older := obj.(*expapi.ThirdPartyResource)
-	older.ResourceVersion = "1"
 	test.TestUpdate(
-		rsrc,
-		existing,
-		older,
+		// valid
+		validNewThirdPartyResource("foo"),
+		func(ctx api.Context, obj runtime.Object) error {
```

Comment:
@nikhiljindal @lavalamp 
I was also thinking about it a bit during the weekend and I think that what we should do is to create "registrytest" framework, which will be just calling the methods from "resttest", but will be aware of all those methods that are specific for etcd. This will also eliminate a bunch of boilerplate for individual tests.
I'm going to make a PR for it (unless you pick it up before me), but once I'm done with refactoring for watch & delete tests.